solr/core/src/java/org/apache/solr/jersey/InjectionFactories.java

```diff
@@ -62,6 +65,25 @@ public SolrQueryResponse provide() {
     public void dispose(SolrQueryResponse instance) {}
   }
 
+  /** Fetch the (existing) SolrCore from the request context */
+  public static class SolrCoreFactory implements Factory<SolrCore> {
```

Comment:
FYI the name of this suggests something it most definitely isn't but I can be okay with it based on it's an inner class of something establishing context.

Reply:
I'd like to keep "Factory" in there because it's implementing a Factory interface, but I agree the overloading of that term is unfortunate.

I can tweak the name so that it's clearer that this doesn't actually construct new SolrCore's though.  That should help I think?